presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/api/ClusterConfigAPI.java

```diff
@@ -0,0 +1,137 @@
+        }
+        return Response.status(ExtendedStatus.MULTI_STATUS).entity(new Gson().toJson(responseList.build())).language("json").build();
+    }
+
+    @POST
+    @Path("/{file}")
+    @ApiOperation(value = "Replace this file with the file at the given URL")
+    @ApiResponses(value = {
+            @ApiResponse(code = 202, message = "Acknowledged request"),
+            @ApiResponse(code = 409, message = "Request conflicts with current state")})
+    public synchronized Response setConfigFileByURL(
+            @PathParam("file") @ApiParam("The name of a file") String file,
+            String url)
+    {
+        ImmutableList.Builder<Response> responseList = ImmutableList.builder();
+        LOGGER.debug("POST /cluster/config/{}", file);
+        for (URI uri : NodeSet.getAllNodes()) {
+            responseList.add(newRequestBuilder(uri.resolve("/config/" + file)).withBody(Entity.entity(url, MediaType.TEXT_PLAIN)).buildPost().invoke());
+        }
+        return Response.status(ExtendedStatus.MULTI_STATUS).entity(new Gson().toJson(responseList.build())).language("json").build();
+    }
+
+    @DELETE
+    @Path("/{file}")
+    @ApiOperation(value = "Delete a configuration file")
+    @ApiResponses(value = {
+            @ApiResponse(code = 202, message = "Acknowledged request"),
+            @ApiResponse(code = 404, message = "Resource not found"),
+            @ApiResponse(code = 409, message = "Request conflicts with current state")})
+    public synchronized Response deleteConfigFile(
+            @PathParam("file") @ApiParam("The name of a file") String file)
+    {
+        ImmutableList.Builder<Response> responseList = ImmutableList.builder();
+        LOGGER.debug("DELETE /cluster/config/{}", file);
+        for (URI uri : NodeSet.getAllNodes()) {
+            responseList.add(newRequestBuilder(uri.resolve("/config/" + file)).buildDelete().invoke());
+        }
+        return Response.status(ExtendedStatus.MULTI_STATUS).entity(new Gson().toJson(responseList.build())).language("json").build();
+    }
+
+    @GET
+    @Path("/{file}/{property}")
+    @Produces({"application/json"})
+    @ApiOperation(value = "Get specific configuration property")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Retrieved property", response = String.class),
+            @ApiResponse(code = 404, message = "Resource not found")})
+    public synchronized Response getConfigProperty(
+            @PathParam("file") @ApiParam("The name of a file") String file,
+            @PathParam("property") @ApiParam("A specific property") String property)
+    {
+        ImmutableList.Builder<Response> responseList = ImmutableList.builder();
+        LOGGER.debug("GET /cluster/config/{}/{}", file, property);
+        for (URI uri : NodeSet.getAllNodes()) {
+            responseList.add(newRequestBuilder(uri.resolve("/config/" + file + "/" + property)).buildGet().invoke());
```

Comment:
Looking at the number of times you have to manually construct URIs in these files, it's probably worthwhile to use `javax.ws.rs.core.UriBuilder`, especially because you are constructing the URIs repeatedly in a loop.

It might look something like this for this particular function.
```java
UriBuilder builder = UriBuilder.fromUri("placeholder/config/").segment(file, property);
for (URI uri : ...) doStuff(builder.uri(uri));
```